app/HutchAgent/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,72 @@
+using HutchAgent.Config;
+using HutchAgent.Services;
+
+namespace HutchAgent.Extensions;
+
+public static class ServiceCollectionExtensions
+{
+  /// <summary>
+  /// Enum for Results Store Providers.
+  /// </summary>
+  enum ResultsStoreProviders
+  {
+    Minio,
+    FileSystem
+  }
+
+  /// <summary>
+  /// Parse the desired Results Store Provider from configuration.
+  /// </summary>
+  /// <param name="c">The configuration object</param>
+  /// <returns>The <c>enum</c> for the chosen Results Store Provider</returns>
+  private static ResultsStoreProviders GetResultsStoreProvider(IConfiguration c)
+  {
+    Enum.TryParse<ResultsStoreProviders>(
+      c["ResultsStore:Provider"],
+      ignoreCase: true,
+      out var resultsStoreProviders);
+
+    return resultsStoreProviders;
```

Comment:
would we need an Exception handling if the configuration value doesn't match any of the enum values?

Reply:
I think it's fine if if crashes on this one. The result store is necessary for hutch so if it errors out, we'd get a reminder that we've forgotten a setting